### DISCUSSION TITLE
Run-time enable/disable of  individual CDC devices?

### DISCUSSION THREAD

**Discussion by MWP** [hathach/tinyusb#2746] [Q&A] [answered]
Hi all,

Is it possible with TinyUSB to enable/disable individual CDC and MSC devices after tud_init() has been called?
cdcd_init() appears to enabled them all at the same time?

I haven't dug too deep on this yet. I'm hoping there is a simple/existing answer.

Thanks in advance!

**Comment by HiFiPhile** [comment, marked as answer]
No it's impossible by the design of USB protocol.

**Reply by MWP** [reply]
Ok.
Is there a method, for example, to set CFG_TUD_CDC=2, and only start one of those CDC on tud_init()?

**Reply by HiFiPhile** [reply]
You need 2 set of configuration descriptors, one with 1 port and another one have 2. Then switch them in tud_descriptor_configuration_cb()
